contrib/db_pools/lib/src/database.rs

```diff
@@ -0,0 +1,117 @@
+use rocket::fairing::{Info, Kind};
+use rocket::futures::future::BoxFuture;
+use rocket::http::Status;
+use rocket::request::{FromRequest, Outcome, Request};
+use rocket::yansi::Paint;
+use rocket::{Build, Ignite, Rocket, Sentinel};
+
+use crate::{Error, Pool};
+
+/// Trait implemented to define a database connection pool.
+pub trait Database: Sized + Send + Sync + 'static {
+    /// The name of this connection pool in the configuration.
+    const NAME: &'static str;
+
+    /// The underlying connection type returned by this pool.
+    /// Must implement [`Pool`].
+    type Pool: Pool;
+
+    /// Returns a fairing that attaches this connection pool to the server.
+    fn fairing() -> Fairing<Self>;
```

Comment:
This `Fairing` type feels pretty weird to me. It seems to me that a `Database` is itself a `Fairing`. Or something like that. There's something not quite right about the setup right now with respect to this fairing, the `Fairing` type, and the `Connection` type.